databricks/resource_databricks_permissions.go

```diff
@@ -0,0 +1,204 @@
+package databricks
+
+import (
+	"fmt"
+	"path"
+	"strconv"
+
+	"github.com/databrickslabs/databricks-terraform/client/model"
+	"github.com/databrickslabs/databricks-terraform/client/service"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func parsePermissionsFromData(d *schema.ResourceData,
+	client *service.DBApiClient) (*model.AccessControlChangeList, string, error) {
+	var objectID string
+	acl := new(model.AccessControlChangeList)
+	for _, mapping := range permissionsResourceIDFields() {
+		v, ok := d.GetOk(mapping.field)
+		if !ok {
+			continue
+		}
+		id, err := mapping.idRetriever(client, v.(string))
+		if err != nil {
+			return nil, "", err
+		}
+		objectID = fmt.Sprintf("/%s/%s", mapping.resourceType, id)
+		err = d.Set("object_type", mapping.objectType)
+		if err != nil {
+			return nil, "", err
+		}
+	}
+	if objectID == "" {
+		return nil, "", fmt.Errorf("At least one type of resource identifiers must be set")
+	}
+	if data, ok := d.GetOk("access_control"); ok {
+		for _, element := range data.([]interface{}) {
+			rawAccessControl := element.(map[string]interface{})
+			change := new(model.AccessControlChange)
+			acl.AccessControlList = append(acl.AccessControlList, change)
+			if v, ok := rawAccessControl["group_name"].(string); ok && v != "" {
+				change.GroupName = &v
+			}
+			if v, ok := rawAccessControl["user_name"].(string); ok && v != "" {
+				change.UserName = &v
+			}
+			if v, ok := rawAccessControl["permission_level"].(string); ok {
+				change.PermissionLevel = v
+			}
+		}
+	}
+	return acl, objectID, nil
+}
+
+func resourcePermissionsCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*service.DBApiClient)
+	acl, objectID, err := parsePermissionsFromData(d, client)
+	if err != nil {
+		return err
+	}
+	err = client.Permissions().AddOrModify(objectID, acl)
+	if err != nil {
+		return err
+	}
+	d.SetId(objectID)
+	return resourcePermissionsRead(d, m)
+}
+
+func resourcePermissionsRead(d *schema.ResourceData, m interface{}) error {
+	id := d.Id()
+	client := m.(*service.DBApiClient)
+	objectACL, err := client.Permissions().Read(id)
+	if err != nil {
+		return err
+	}
+	for _, mapping := range permissionsResourceIDFields() {
+		if mapping.objectType != d.Get("object_type").(string) {
+			continue
+		}
+		identifier := path.Base(id)
+		err := d.Set(mapping.field, identifier)
+		if err != nil {
+			return errors.Wrapf(err,
+				"Cannot set mapping field %s to %s",
+				mapping.field, id)
+		}
+		break
+	}
+	acl := objectACL.ToAccessControlChangeList()
+	me, err := client.Users().Me()
+	if err != nil {
+		return errors.Wrapf(err, "Cannot self-identify")
+	}
+	accessControl := acl.AccessControl(me.UserName)
+	err = d.Set("access_control", accessControl)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
This should set the acl Id to "" if the read returns nothing, or if the parent resource is deleted i.e cluster permissions, if the cluster is deleted the permissions should be treated as deleted so state issues will not occur.

Reply:
done